File: src/main/java/com/footverse/user/dto/UserResponse.java
package com.footverse.user.dto;

import java.time.LocalDateTime;

import com.footverse.user.entity.Role;

/**
 * A user's profile returned to clients. Passwords are never included.
 *
 * @param id        the user id
 * @param email     the email
 * @param fullName  the full name
 * @param phone     the phone number
 * @param avatarUrl the avatar URL, if any
 * @param role      the account role
 * @param enabled   whether the account is enabled
 * @param createdAt the creation timestamp
 * @param updatedAt the last-update timestamp
 */
public record UserResponse(
        Long id,
        String email,
        String fullName,
        String phone,
        String avatarUrl,
        Role role,
        boolean enabled,
        LocalDateTime createdAt,
        LocalDateTime updatedAt) {
}

File: admin/src/features/auth/models/user-response.ts
import type { Role } from './role';

/**
 * A user's profile (dto-spec §7). Passwords are never included. Mirrors the
 * frozen DTO field-for-field.
 */
export interface UserResponse {
  readonly id: number;
  readonly email: string;
  readonly fullName: string;
  readonly phone: string;
  readonly avatarUrl?: string;
  readonly role: Role;
  readonly enabled: boolean;
  readonly createdAt: string;
  readonly updatedAt: string;
}
